crates/notion-core/src/image/mod.rs

```diff
@@ -66,6 +66,20 @@ impl System {
         Ok(new_path)
     }
 
+    /// Reproduces the Notion-enabled `PATH` environment variable for situations where
+    /// Notion has been deactivated
+    pub fn enabled_path() -> Fallible<OsString> {
+        let old_path = envoy::path().unwrap_or(envoy::Var::from(""));
+        let shim_dir = path::shim_dir()?;
+
+        let mut parts = old_path.split();
+
+        if !old_path.split().any(|part| part == shim_dir) {
```

Comment:
No need to split twice, you can simplify this a little by doing:

```rust
if !parts.any(|part| part == shim_dir) {
```

Reply:
My understanding was that `any()` was one of the methods that consumed the iterator, so if we did `parts.any(...)` then later on the `parts.join()` call would fail / not include all of the items in the iterable.

Reply:
Confirmed, switching to `if !parts.any(...)` causes the tests to fail, as the `any` consumes part of the iterator, leaving the `parts.join()` to only return part of the `PATH`, instead of everything that we want it to.

Reply:
This might be slightly cleaner:

```rust
        if !old_path.split().any(|part| part == shim_dir) {
            old_path.split().prefix(vec![shim_dir]).join().unknown()?
        } else {
            old_path
        }
```

The double split smells to me like the `envoy` API isn't quite ideal, but I'm not exactly sure what the right thing is. For now, splitting twice seems unavoidable.

Reply:
Updated